### PR TITLE
Replace String::remove() by a makeStringByRemoving() free function

### PR DESCRIPTION
**Pull request by cdumez**
#### 13261d6a5062c32df336b4b4a687c62cb1c4fe77
<pre>
Replace String::remove() by a makeStringByRemoving() free function
<a href="https://bugs.webkit.org/show_bug.cgi?id=239995">https://bugs.webkit.org/show_bug.cgi?id=239995</a>

Reviewed by Yusuke Suzuki.

Replace String::remove() by a makeStringByRemoving() free function. This is a
step towards making String immutable.

* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm:
(WebKit::lastCNAMEDomain):
* Source/WebKit/UIProcess/Inspector/win/InspectorResourceURLSchemeHandler.cpp:
(WebKit::InspectorResourceURLSchemeHandler::platformStartTask):
* Source/WTF/wtf/URL.cpp:
(WTF::URL::remove):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::makeStringByRemoving):
(WTF::String::removeInternal): Deleted.
(WTF::String::remove): Deleted.
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/Modules/mediastream/RTCDTMFSender.cpp:
(WebCore::RTCDTMFSender::playNextTone):
* Source/WebCore/dom/CharacterData.cpp:
(WebCore::CharacterData::deleteData):
* Source/WebCore/dom/FragmentDirectiveParser.cpp:
(WebCore::FragmentDirectiveParser::parseFragmentDirective):
* Source/WebCore/dom/Range.cpp:
(WebCore::processContentsBetweenOffsets):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::CompositeEditCommand::deleteInsignificantText):
* Source/WebCore/inspector/InspectorOverlayLabel.cpp:
(WebCore::InspectorOverlayLabel::draw):
* Source/WebCore/inspector/InspectorStyleSheet.cpp:
(WebCore::InspectorStyleSheet::deleteRule):
* Source/WebCore/platform/graphics/gstreamer/mse/GStreamerMediaDescription.cpp:
(WebCore::GStreamerMediaDescription::extractCodecName):
* Source/WebCore/platform/network/curl/CookieJarDB.cpp:
(WebCore::CookieJarDB::deleteCookie):
* Source/WebCore/platform/text/win/LocaleWin.cpp:
(WebCore::LocaleWin::shortTimeFormat):

Canonical link: <a href="https://commits.webkit.org/250239@main">https://commits.webkit.org/250239@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@293760">https://svn.webkit.org/repository/webkit/trunk@293760</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

**Comment by cdumez** [comment]
Patch is ready for review.

File: Source/WebCore/inspector/InspectorOverlayLabel.cpp [hunk 1]
```diff
@@ -239,7 +239,7 @@ Path InspectorOverlayLabel::draw(GraphicsContext& context, float maximumLineWidt
                 text = makeString(text, ellipsis);
                 while (currentLineWidth + textWidth + (labelPadding * 2) > maximumLineWidth && text.length() > 1) {
                     // Remove the second from last character (the character before the ellipsis) and remeasure.
-                    text.remove(text.length() - 2);
+                    text = text.left(text.length() - 2);
```
Comment (Constellation) [inline Source/WebCore/inspector/InspectorOverlayLabel.cpp]: Is it correct? I think the original code was only removing one second-from-last character (the last character is remaining).

**Review by Constellation — CHANGES_REQUESTED** [review]

File: Source/WebCore/inspector/InspectorOverlayLabel.cpp (hunk 1 above)
Comment (cdumez) [reply Source/WebCore/inspector/InspectorOverlayLabel.cpp]: Oh, you're right, I am removing the 2 last characters and the previous code was removing only the second to last. Will fix.

**Review by Constellation — APPROVED** [review]
r=me

**Pull request closed without merging**